app/domain/repository/persist_repository_from_upstream.rb

```diff
@@ -0,0 +1,129 @@
+# frozen_string_literal: true
+
+class Repository::PersistRepositoryFromUpstream
+  def self.create_or_update_from_host_data(upstream_raw_host_data)
+    return unless upstream_raw_host_data
+
+    ActiveRecord::Base.transaction do
+      existing_repo = find_repository_from_host_data(upstream_raw_host_data)
+
+      if existing_repo.nil?
+        new_repository = Repository.new(uuid: upstream_raw_host_data.repository_uuid, full_name: upstream_raw_host_data.full_name)
+
+        new_repository.host_type = upstream_raw_host_data.host_type
```

Comment:
🎱 Can these two methods get consolidated?

```
def create_or_update
  repo = find_repo || repo.new(..)

  repo.update(
     host_type: upstream_raw_host_data.host_type
     license: upstream_raw_host_data.formatted_license
  )
end
```

Reply:
I kept them separate because that is how it was in the original code. Putting them in one implies `find_repo` would always find the right one instead of having the target repository passed in directly. Right now the only time we have a specific `Repository` being passed in is via `Repository.update_from_repository()` which is called in `Repository.update_all_info()` and in the name clash logic in this class. Maybe I need to try and run some queries to figure out if we do have name clashes in the database that need to be handled. It would be nice to implement your suggestion here.